app/views/homepage/index.html.erb

```diff
@@ -21,21 +21,83 @@
           <p class="home-top__intro home-top__intro--simpler"><%= t('homepage.index.intro_simpler') %></p>
 
           <form action="/search" method="get" role="search">
-            <%= render "govuk_publishing_components/components/search", on_govuk_blue: true %>
+            <%= render "govuk_publishing_components/components/search", {
+              on_govuk_blue: true,
+              data_attributes: {
+                track_category: "homepageClicked",
+                track_action: "searchSubmitted",
+                track_label: "/search/all",
+                track_dimension_index: 29,
+                track_dimension: "Search on GOV.UK",
+              },
+            } %>
           </form>
         </div>
 
         <div class="govuk-grid-column-one-third">
-          <div class="home-top__links">
+          <div class="home-top__links" data-module="gem-track-click">
             <h2 class="home-top__links-title"><%= t('homepage.index.popular') %></h2>
+            <%
+              link_class = "govuk-link govuk-link--inverse home-top__link-item"
+              items = []
+              items[0] = link_to(t("homepage.index.covid"),
```

Comment:
Personal prefs thing, `items << link_to...` instead of `items[n] = link_to...` would anonymise the array keys in this separated declaration. I wonder if this would reduce the tiny little bit of extra cognisance of adding an extra number to the items array as well as writing a new list item if someone did need to add stuff to the list.

Reply:
I did think about that but I wanted to ensure order of the links. I know I could order the pushes in the code that way... 🤔

Reply:
Actually you're right it'd be easier to maintain. I thought being explicit would make it clearer about the order. Have amended.